### DISCUSSION TITLE
Override colors in built-in colorscheme

### DISCUSSION THREAD

**Discussion by damanis** [Iron-E/nvim-highlite#34] [Q&A]
Is there a way to override, for example, only `bg` in built-in colorschemes without auto command?
```
local Highlite = require('highlite')
local palette = Highlite.palette('iceberg')
vim.api.nvim_create_autocmd('ColorScheme', {
  callback = function()
    vim.api.nvim_set_hl(0, 'Normal', {fg = palette.fg, bg = '#07111e'})
  end,
  group = vim.api.nvim_create_augroup('config', {clear = true}),
  pattern = 'highlite*',
})
```

**Comment by Iron-E** [comment]
Great question!

Yes, an `autocmd` is probably the best way to do this. I would do it like this:

```lua
vim.api.nvim_create_autocmd('ColorScheme', {
    callback = function()
        vim.cmd.highlight 'Normal guibg=#07111e'
        -- other groups
    end,
    group = vim.api.nvim_create_augroup('config', {}),
    pattern = 'highlite*',
})
```

Unlike `nvim_set_hl` (which replaces highlight groups), `:highlight` patches highlight groups with the given arguments. So, you can use it to just alter one part of a group and leave the rest intact.

**Comment by damanis** [comment]
Thank you, will try. I asked, because usually themes allows to change theme's palette and colour groups.
Very good plugin.
